notificationdemo/source/java/src/org/kaaproject/kaa/demo/notification/NotificationDemo.java

```diff
@@ -51,74 +58,107 @@ public static void main(String[] args) {
         // Add a notification listener that listens to all notifications.
         kaaClient.addNotificationListener(new NotificationListener() {
             @Override
-            public void onNotification(long id, SampleNotification sampleNotification) {
-                LOG.info("Notification for topic id [{}] received.", id);
-                LOG.info("Notification body: {}", sampleNotification.getMessage());
+            public void onNotification(long id, SecurityAlert sampleNotification) {
+                LOG.info("Notification for topic id [{}] and name [{}] received.", id, getTopic(id).getName());
+                LOG.info("Notification body: {} \n", sampleNotification.getAlertMessage());
+                LOG.info("Notification alert type: {} \n", sampleNotification.getAlertType());
             }
         });
 
         // Start the Kaa client and connect it to the Kaa server.
         kaaClient.start();
 
         // Get available notification topics.
-        List<Topic> topicList = kaaClient.getTopics();
+        topics = kaaClient.getTopics();
 
         // List the obtained notification topics.
-        showTopicList(topicList);
-
-        try {
-            
-            // Wait for some input before exiting.
-            System.in.read();
-        } catch (IOException e) {
-            LOG.error("IOException was caught", e);
+        showTopics();
+
+        LOG.info("Type topic ID in order to subscribe to ones or type any text to exit");
```

Comment:
Create a new method, for example, showEnteropicIDMessage(), move this line into this method and call it after every operation in code like topic list update, after subscription on some topic.

This line printed only once and could be lost in a mass of logs. Highlight this log message with new line characters (\n).